File: kingbase/src/test/java/com/dbx/agent/kingbase/KingbaseAgentTest.java
package com.dbx.agent.kingbase;

import com.dbx.agent.DatabaseAgent;
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class KingbaseAgentTest extends JdbcFakeExecutionBehaviorTest {
    @Override
    protected DatabaseAgent createAgent() {
        return new KingbaseAgent();
    }

    @Override
    protected String resultSetSql() {
        return "CALL sample_proc()";
    }

    @Test
    void declaresKingbasePostgresLikeProfile() {
        KingbaseAgent agent = new KingbaseAgent();

        Assertions.assertEquals("com.kingbase8.Driver", agent.getProfile().getDriverClass());
        Assertions.assertEquals("jdbc:kingbase8://{host}:{port}/{database}", agent.getProfile().getUrlTemplate());
    }
}

File: kingbase/src/test/kotlin/com/dbx/agent/kingbase/KingbaseAgentTest.kt
package com.dbx.agent.kingbase

import com.dbx.agent.DatabaseAgent
import com.dbx.agent.test.JdbcFakeExecutionBehaviorTest
import kotlin.test.Test
import kotlin.test.assertEquals

class KingbaseAgentTest : JdbcFakeExecutionBehaviorTest() {
    override fun createAgent(): DatabaseAgent {
        return KingbaseAgent()
    }

    override fun resultSetSql(): String = "CALL sample_proc()"

    @Test
    fun `declares kingbase postgres-like profile`() {
        val agent = KingbaseAgent()

        assertEquals("com.kingbase8.Driver", agent.profile.driverClass)
        assertEquals("jdbc:kingbase8://{host}:{port}/{database}", agent.profile.urlTemplate)
    }
}
